integration_tests/test_gravity.py

```diff
@@ -606,3 +606,96 @@ def check_cronos_balance_change():
+        erc20 = deploy_contract(
+            geth,
+            CONTRACTS["TestBlackListERC20"],
+            (ADDRS["signer1"],),
+        )
+
+        balance = erc20.caller.balanceOf(ADDRS["validator"])
+        assert balance == 100000000000000000000000000
+        amount = 1000
+
+        print("send to cronos crc20")
+        recipient = HexBytes(ADDRS["community"])
+        txreceipt = send_to_cosmos(
+            gravity.contract, erc20, recipient, amount, KEYS["validator"]
+        )
+        assert txreceipt.status == 1, "should success"
+        assert erc20.caller.balanceOf(ADDRS["validator"]) == balance - amount
+
+        denom = f"gravity{erc20.address}"
+        crc21_contract = None
+
+        def local_check_auto_deployment():
+            nonlocal crc21_contract
+            crc21_contract = check_auto_deployment(
+                cli, denom, cronos_w3, recipient, amount
+            )
+            return crc21_contract
+
+        wait_for_fn("send-to-crc21", local_check_auto_deployment)
+
+        # get voucher nonce
+        old_nonce = gravity.contract.caller.state_lastRevertedNonce()
+        old_balance1 = erc20.caller.balanceOf(ADDRS["signer1"])
+
+        # send it back to blacklisted address
+        tx = crc21_contract.functions.send_to_chain(
+            ADDRS["signer1"], amount, 0, 1
+        ).buildTransaction({"from": ADDRS["community"]})
+        txreceipt = send_transaction(cronos_w3, tx, KEYS["community"])
+        assert txreceipt.status == 1, "should success"
+
+        def check():
+            nonce = gravity.contract.caller.state_lastRevertedNonce()
+            return old_nonce + 1 == nonce
+
+        wait_for_fn("send-to-ethereum", check)
+
+        # check that voucher has been created
+        voucher = gravity.contract.caller.state_RevertedVouchers(old_nonce)
+        assert voucher[0] == erc20.address
+        assert voucher[1] == ADDRS["signer1"]
+        assert voucher[2] == amount
+
+        # check balance is the same
+        new_balance1 = erc20.caller.balanceOf(ADDRS["signer1"])
+        assert old_balance1 == new_balance1
+
+        old_balance2 = erc20.caller.balanceOf(ADDRS["signer2"])
+
+        # try to redeem voucher with non recipient address
+        with pytest.raises(Exception):
+            gravity.contract.functions.redeemVoucher(
+                old_nonce, ADDRS["signer2"]
+            ).buildTransaction({"from": ADDRS["validator"]})
+
+        # send user1 some fund for gas
+        send_transaction(
+            geth, {"to": ADDRS["signer1"], "value": 10**17}, KEYS["validator"]
+        )
+        # redeem voucher
+        tx = gravity.contract.functions.redeemVoucher(
+            old_nonce, ADDRS["signer2"]
+        ).buildTransaction({"from": ADDRS["signer1"]})
+        txreceipt = send_transaction(geth, tx, KEYS["signer1"])
+        assert txreceipt.status == 1, "should success"
+        wait_for_new_blocks(cli, 1)
```

Comment:
I think you intent to wait for new block on geth, there's a `wait_for_new_blocks_w3` for that.